sasdata/data_util/manipulations.py

```diff
@@ -63,19 +63,18 @@ def get_q_compo(dx: float, dy: float, detector_distance: float, wavelength: floa
     return out
 
 
-def flip_phi(phi: float) -> float:
+def normalize_angle(phi: float, closed_at_end: bool = False) -> float:
     """
-    Force phi to be within the 0 <= to <= 2pi range by adding or subtracting
-    2pi as necessary
+    Force phi to be within the half-open interval 0 <= phi < 2pi, unless
+    closed_at_end is set to True, in which case the interval is 0 < phi <= 2pi.
 
-    :return: phi in >=0 and <=2Pi
+    :param phi: angle to return in a normalized form (0 <= phi < 2pi)
+    :param closed_at_end: if True, changes the interval to 0 < phi <= 2pi
     """
-    if phi < 0:
-        phi_out = phi + (2 * math.pi)
-    elif phi > (2 * math.pi):
-        phi_out = phi - (2 * math.pi)
-    else:
-        phi_out = phi
+    phi_out = phi % (2 * np.pi)
+    if closed_at_end and phi_out == 0:
+        phi_out = 2 * np.pi
+
```

Comment:
Clever way to replace the if/elif, if with a single line! However the closed at end seems clunky.  Am I correct in assuming the problem is that the modulus of 2pi/2pi is 0 is the problem? If so would using a `divmod(a,b)` operation and making the if statement be `if (b=0 and a>0) phi_out=2*np.pi` simplify things and not require the new `closed_at_end` variable which changes the calling interface to the function?

Reply:
I've implemented `divmod` as part of a further revision to this function.